### DISCUSSION TITLE
How do you open directories?

### DISCUSSION THREAD

**Discussion by KaBankz** [NvChad/NvChad#1670] [Q&A] [answered]
Hey, I just installed NvChad and was having trouble opening directories.

Doing `nvim ~/.config` or `nvim .` does not show a directory listing like in `netrw`, just a blank buffer. I can see that NvChad uses `nvim-tree` instead of `netrw`, but, I tried looking around the NvChad documentation and I couldn't find anything mentioning how to open directories in NvChad. I just want to get a directory listing when opening a directory.

So, how do you open directories using NvChad?

Thanks in advance

**Comment by siduck** [comment, marked as answer]
@KaBankz nvim . doesnt work cuz nvimtree is lazyloaded, to skip lazyloading you can just load it at VImEnter event, then do packercompile

**Reply by KaBankz** [reply]
Thanks for the help, but I'm not sure how to do this. I'm new to lua, nvim and NvChad, so apologies for the lack of understanding. I assume it's something like `autocmd("VimEnter", ...` added to `custom/init.lua` but I am not sure. If you could show me what exactly I have to add and where to add it, I would greatly appreciate it.

**Reply by siduck** [reply]
@KaBankz please read our plugin docs

**Reply by KaBankz** [reply]
Thanks! I can't believe I missed that. For anyone wondering how I did it, I just created `lua/custom/chadrc.lua` and put this in it.

```lua
local M = {}

M.plugins = {
  ["kyazdani42/nvim-tree.lua"] = {
    event = "VimEnter",
  },
}

return M
```

**Reply by griff-rees** [reply]
This also applies to opening any *folder* from [command-line mode](https://vimdoc.sourceforge.net/htmldoc/cmdline.html), which I've found confusing in the default configuration. Thanks very much.

**Reply by rayhagimoto** [reply]
slightly different solution based on the updated [docs](https://nvchad.com/docs/config/plugins):

In `custom/chadrc.lua`

```lua
--  ~/.config/nvim/lua/custom/chadrc.lua
local M = {}

M.plugins = "custom.lua"

return M
```

in `custom/plugins.lua`:

```lua
--  ~/.config/nvim/lua/custom/plugins.lua
local plugins = {

  -- copy the nvim-tree/nvim-tree.lua entry from "~/.config/nvim/lua/plugins/init.lua"
  -- but add the line "lazy = false"
  {
    "nvim-tree/nvim-tree.lua",
    lazy = false,
    cmd = { "NvimTreeToggle", "NvimTreeFocus" },
    init = function()
      require("core.utils").load_mappings "nvimtree"
    end,
    opts = function()
      return require "plugins.configs.nvimtree"
    end,
    config = function(_, opts)
      dofile(vim.g.base46_cache .. "nvimtree")
      require("nvim-tree").setup(opts)
      vim.g.nvimtree_side = opts.view.side
    end,
  },

}

return plugins
```

**Reply by siduck** [reply]
@rayhagimoto  if you want it  on start then just add this code in your custom/init.lua

```lua
vim.schedule(function()
  require("nvim-tree.api").tree.open()
end)
```

**Reply by griff-rees** [reply]
Thanks @siduck, that *mostly* sorts it for me. However: now `nvim-tree` switches on at start even if simply opening a file (rather than a directly). Is there a modification that only applies when opening a folder path?

**Reply by griff-rees** [reply]
@siduck 

Following your suggestion and other `nvtree` [suggestions](https://github.com/nvim-tree/nvim-tree.lua/wiki/Open-At-Startup) I'm using this config in my `custom/init.lua`, which opens `nvtree` if passed a directory (in which case it changes to that directory), or `.` or no path on startup (staying in the directory called from):

```lua
local autocmd = vim.api.nvim_create_autocmd

local function open_nvim_tree(data)

    -- buffer is a directory
    local directory = vim.fn.isdirectory(data.file) == 1

    -- buffer is a [No Name]
    local no_name = data.file == "" and vim.bo[data.buf].buftype == ""

    if not directory and not no_name then
        return
    end

    if directory then
        -- change to the directory
        vim.cmd.cd(data.file)
    end

    -- open the tree
    require("nvim-tree.api").tree.open()
end

autocmd({ "VimEnter" }, { callback = open_nvim_tree })
```

**Reply by siduck** [reply]
@griff-rees looks good!

**Reply by griff-rees** [reply]
Thanks: appreciate the encouragement. It's a great config structure to learn from.
